src/olympia/addons/tests/test_indexers.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+from olympia.amo.tests import TestCase
+from olympia.addons.models import (
+    Addon, attach_categories, attach_tags, attach_translations)
+from olympia.addons.indexers import AddonIndexer
+
+
+class TestAddonIndexer(TestCase):
+    fixtures = ['base/users', 'base/addon_3615']
+
+    def setUp(self):
+        super(TestAddonIndexer, self).setUp()
+        self.attrs = (
+            'id', 'slug', 'created', 'default_locale', 'last_updated',
+            'weekly_downloads', 'average_daily_users', 'status', 'type',
+            'hotness', 'is_disabled', 'is_listed',
+        )
+        self.transforms = (attach_categories, attach_tags, attach_translations)
+        self.indexer = AddonIndexer()
+
+    def _extract(self):
+        qs = Addon.objects.filter(id__in=[3615])
+        for t in self.transforms:
+            qs = qs.transform(t)
+        self.addon = list(qs)[0]
+        return self.indexer.extract_document(self.addon)
+
+    def test_extract_attributes(self):
+        extracted = self._extract()
+        for attr in self.attrs:
+            assert extracted[attr] == getattr(self.addon, attr)
```

Comment:
Can we go the other way around? Iterate through extracted and make sure it's in `self.attrs`? Or add some sanity checks based on my suggestion in https://github.com/mozilla/addons-server/pull/1961/files#r57228298